crates/tracing-instrumentation/src/exporter.rs

```diff
@@ -135,3 +146,67 @@ impl<T: SpanExporter + 'static> SpanExporter for ResourceModifyingSpanExporter<T
         self.resource.store(Arc::new(resource.clone()))
     }
 }
+
+#[derive(Debug)]
+pub(crate) struct RuntimeModifierSpanExporter<E>
+where
+    E: SpanExporter,
+{
+    inner: E,
+    resources: Resource,
+    injected: bool,
+}
+
+impl<E> RuntimeModifierSpanExporter<E>
+where
+    E: SpanExporter,
+{
+    pub fn new(inner: E) -> Self {
+        Self {
+            inner,
+            resources: Resource::empty(),
+            injected: false,
+        }
+    }
+}
+
+impl<E> SpanExporter for RuntimeModifierSpanExporter<E>
+where
+    E: SpanExporter,
+{
+    fn export(
+        &mut self,
+        batch: Vec<SpanData>,
+    ) -> BoxFuture<'static, opentelemetry_sdk::export::trace::ExportResult> {
+        if !self.injected && GLOBAL_NODE_ID.initialized() {
+            self.injected = true;
+            let resources = self.resources.clone();
+            let resources = resources.merge(&Resource::new(vec![KeyValue::new(
```

Comment:
I think you can write `self.resources.merge` w/o the clone.

Reply:
Yes, i have this change locally but did not push it yet :)

Reply:
I also added a feature `create_per_service` which allows us to switch to the other vide of traces (as shown in the example below). This comes with the drawback that all `crates` of the system now show up as their own service on `jaeger` this is why this is feature gated